e3db/nist_crypto.py

```diff
@@ -0,0 +1,93 @@
+from base_crypto import BaseCrypto
+import os
+
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives import serialization
+from cryptography.hazmat.primitives.asymmetric import ec
+from cryptography.hazmat.primitives.ciphers.aead import AESGCM
+from cryptography.hazmat.primitives.kdf.hkdf import HKDF
+
+
+class NistCrypto(BaseCrypto):
+
+    @classmethod
+    def encrypt_secret(self, ak, plain, nonce):
+        # Prepend the nonce to the encrypted value for parity with the Sodium implementation
+        return nonce + AESGCM(ak).encrypt(nonce, plain, None)
+
+    @classmethod
+    def decrypt_secret(self, ak, ciphertext, nonce):
+        return AESGCM(ak).decrypt(nonce, ciphertext, None)
+
+    @classmethod
+    def encrypt_ak(self, private_key, public_key, ak, nonce):
+        # Prepend the nonce to the encrypted value for parity with the Sodium implementation
+        return nonce + self.exchange(private_key, public_key).encrypt(nonce, ak, None)
+
+    @classmethod
+    def decrypt_eak(self, private_key, public_key, eak, nonce):
+        return self.exchange(private_key, public_key).decrypt(nonce, eak, None)
+
+    @classmethod
+    def exchange(self, private_key, public_key):
```

Comment:
Probably this should have an underscore prefix (`_exchange`).

Reply:
:+1: